rpc/rpc-codegen/src/main/java/com/strategyobject/substrateclient/rpc/codegen/encoder/EncoderCompositor.java

```diff
@@ -48,6 +45,30 @@ public EncoderCompositor(@NonNull ProcessorContext context,
         this.scaleRegistryVarName = scaleRegistryVarName;
     }
 
+    private CodeBlock getNonGenericCodeBlock(TypeMirror type) {
+        return CodeBlock.builder()
+                .add("$T.$L(($T) ", EncoderPair.class, PAIR_FACTORY_METHOD, RpcEncoder.class)
+                .add("$L.resolve($T.class)",
+                        encoderRegistryVarName,
+                        context.isAssignable(type, selfEncodable) ?
+                                selfEncodable :
+                                type)
+                .add(", ($T) ", ScaleWriter.class)
+                .add("$L.resolve($T.class)", scaleRegistryVarName, type)
+                .add(")")
+                .build();
+    }
+
+    private CodeBlock getArrayNonGenericCodeBlock(ArrayType type) {
```

Comment:
I think the method isn't necessary and the method `getNonGenericCodeBlock` covers this case.

Reply:
+